### PR TITLE
system-variables: add tiflash_hashagg_preaggregation_mode

### PR DESCRIPTION
**Pull request by guo-shaoge**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18226
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -5866,6 +5866,19 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
...
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
+- Type: Enumeration
+- Default value: `force_preagg`
+- Value options: `force_preagg`, `force_streaming`, `auto`
+- This variable controls which pre-aggregation strategy is used in the first stage of a two-stage or three-stage HashAgg pushed down to TiFlash:
+    - `force_preagg`: TiFlash forces pre-aggregation in the first stage of HashAgg, same with behavior prior to version v8.3.0.
+    - `force_streaming`: TiFlash directly sends data to the next stage of HashAgg without pre-aggregation.
+    - `auto`: TiFlash automatically selects whether to perform pre-aggregation based on the current workload's aggregation degree.
```
Comment (Oreoxmt) [inline system-variables.md:5880]:
```suggestion
- This variable controls the pre-aggregation strategy used in the first stage of two-stage or three-stage HashAgg operations pushed down to TiFlash:
    - `force_preagg`: TiFlash forces pre-aggregation in the first stage of HashAgg. This behavior is consistent with the behavior before v8.3.0.
    - `force_streaming`: TiFlash directly sends data to the next stage of HashAgg without pre-aggregation.
    - `auto`: TiFlash automatically selects whether to perform pre-aggregation based on the current workload's aggregation degree.
```

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 2]
```diff
@@ -5866,6 +5866,19 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
...
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
+- Type: Enumeration
+- Default value: `force_preagg`
+- Value options: `force_preagg`, `force_streaming`, `auto`
+- This variable controls the pre-aggregation strategy used in the first stage of two-stage or three-stage HashAgg operations pushed down to TiFlash:
+    - `force_preagg`: TiFlash forces pre-aggregation in the first stage of HashAgg. This behavior is consistent with the behavior before v8.3.0.
+    - `force_streaming`: TiFlash directly sends data to the next stage of HashAgg without pre-aggregation.
+    - `auto`: TiFlash automatically selects whether to perform pre-aggregation based on the current workload's aggregation degree.
```
Comment (qiancai) [inline system-variables.md:5880]:
```suggestion
    - `auto`: TiFlash automatically chooses whether to perform pre-aggregation based on the current workload's aggregation degree.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: system-variables.md [hunk 3]
```diff
@@ -5866,6 +5866,19 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
...
 
+### tiflash_hashagg_preaggregation_mode <span class="version-mark">New in v8.3.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
+- Type: Enumeration
+- Default value: `force_preagg`
+- Value options: `force_preagg`, `force_streaming`, `auto`
+- This variable controls the pre-aggregation strategy used in the first stage of two-stage or three-stage HashAgg operations pushed down to TiFlash:
```
Comment (qiancai) [inline system-variables.md:5877]:
```suggestion
- This variable controls the pre-aggregation strategy used during the first stage of two-stage or three-stage HashAgg operations pushed down to TiFlash:
```

**Pull request merged**